language/move-prover/src/prelude.bpl

```diff
@@ -802,6 +802,12 @@ procedure {:inline 1} $AddU64(src1: $Value, src2: $Value) returns (dst: $Value)
     dst := $Integer(i#$Integer(src1) + i#$Integer(src2));
 }
 
+procedure {:inline 1} $AddU64_unchecked(src1: $Value, src2: $Value) returns (dst: $Value)
```

Comment:
I suppose we could reduce the amount of code by having a boolean flag about whether to check or not?

Reply:
Then we have more code which the prover need to evaluate (I don't think Boogie does constant folding).